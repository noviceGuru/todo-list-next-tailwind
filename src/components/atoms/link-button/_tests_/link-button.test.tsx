import { render, screen } from "@testing-library/react"
import '@testing-library/jest-dom'

import LinkButton from "../link-button"

test('link-button renders correctly, reproduces correct link and text', () => {
	render(<LinkButton
		text="link button dummy text"
		to="pages/a-dummy-url"
	/>)

	const linkButton = screen.getByRole('link', {
		name: /link button dummy text/i
	  })

	expect(linkButton).toBeInTheDocument()
	expect(linkButton.textContent).toBe("link button dummy text")
	expect(linkButton).toMatchSnapshot()
})