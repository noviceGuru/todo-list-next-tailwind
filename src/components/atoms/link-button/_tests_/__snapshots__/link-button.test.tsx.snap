// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link-button renders correctly, reproduces correct link and text 1`] = `
<a
  href="pages/a-dummy-url"
>
  link button dummy text
</a>
`;
